backend/whisprer.py

```diff
@@ -0,0 +1,100 @@
+import diart.operators as dops
+import rx.operators as ops
+from utils import concat, jsonify_transcription
+import traceback
+from client import Client
+import asyncio
+import threading
+import logging
+
+
+class Whisprer:
+
+    def __init__(self):
+        self.clients = {}
+
+    def handle_new_chunks(self, sid):
+        logging.info("New chunks handler started")
+        client = self.clients[sid]
+        source = client.get_source()
+        while True:
+            if not client.audio_chunks.empty():
+                current_chunk = client.audio_chunks.get()
+                if current_chunk is None:
+                    client.complete_stream()
+                    return
+                # not a heavy operation but a blocking one at times, shouldn't block the main thread thanks to asyncio
+                source.receive_chunk(current_chunk)
+
+    # FULL CREDIT TO JUANMA CORIA FOR THIS DIART IMPLEMENTATION WITH WHISPER (COLOR YOUR CAPTIONS ON MEDIUM)
+    def transcription_thread(self, sid):
+        logging.info("Transcription thread started")
+
+        client = self.clients[sid]
+        source = client.get_source()
+        diarization_pipeline = client.get_diarization_pipeline()
+        config = client.get_pipeline_config()
+        asr = client.get_transcriber()
+
+        # Split the stream into 2s chunks for transcription
+        transcription_duration = 2
```

Comment:
Should be configurable?

Reply:
True, I wouldn't tweak it for now. I need some additional clarity on how this works behind the scenes to understand the product of giving the transcription duration variable different values.

Reply:
Gotcha